### PR TITLE
Add frr-reload.py to FRR container

### PR DESCRIPTION
**Pull request by mslocrian**
**- What I did**
I have added frr-reload.py to the frr docker container.

For those who don't use it, it is a great tool for those who want to deploy their own frr configuration to the sonic host. You can do the following:
1. push an updated frr.conf to /etc/sonic/frr/frr.conf
1. ensure your frr.conf is functional - `docker exec -ti bgp vtysh -f /etc/frr/frr.conf --dryrun`
1. run `docker exec -ti bgp /usr/bin/frr-reload.py --reload /etc/frr/frr.conf` 

frr-reload.py makes a local copy of running config, and compares it to the provided config. It then only applies the diffs to vtysh. It's much nicer than pushing a config and having to reload a container (in the majority of cases)

Note, I haven't actually rebuilt the sonic image and tested this properly, but I shall! 

**- How I did it**
Copied frr-reload.py from FRR repo into frr container and modified Dockerfile.j2

**- How to verify it**

**- Description for the changelog**
Added frr-reload.py to FRR container.

**- A picture of a cute animal (not mandatory but encouraged)**
Animal

### CODE REVIEW

**Comment by mslocrian** [comment]
Erm. Accidentally have the daemons and daemons.conf changes in here too.

**Comment by mslocrian** [comment]
Rebased! Should have branched the other PR of mine.

**Comment by nikos-github** [comment]
I believe this is part of the pythontools debian pkg that is built along with frr. Perhaps it is best for those who need it, to install the pkg rather than bring it into sonic-buildimage as it then becomes difficult to track and reflect any changes to it from its source location. Sonic team can comment here @lguohan

**Comment by mslocrian** [comment]
> I believe this is part of the pythontools debian pkg that is built along with frr. Perhaps it is best for those who need it, to install the pkg rather than bring it into sonic-buildimage as it then becomes difficult to track and reflect any changes to it from its source location. Sonic team can comment here @lguohan

Ah, you're right: https://github.com/FRRouting/frr/releases. frr-pythontools*.deb

Personally, if people are going to be using the features that you've added to allow to split frr configuration, then this would be a super helpful utility to just have on hand. 

That being said, if someone is going to make the conscious decision to enable the split frr functionality, then I suppose that they can also just install it.

So, I suppose that there are two questions.
1. Do the powers to be think it's a helpful enough utility to include?
1. If so, would that be something to be added to `docker_fpm_frr_debs` (wherever that is)?

**Comment by mslocrian** [comment]
@nikos-github, I've shuffled things around so I'm not COPY'ing the file directly, but installing the frr-pythonutils package which is built when the frr package is built. 

This should keep things consistent for when one upgrades frr. 

I suppose the discussion of whether or not it should be included in the base image should still be had, but it's a slick utility for those wanting to externally manage frr.conf.

**Comment by mslocrian** [comment]
Verified change works in fresh image install.

```
root@sonic:/home/sonic# docker exec -t -i bgp ls /usr/lib/frr
babeld	eigrpd	       frrcommon.sh  ldpd    ospfd  ripd     watchfrr
bfdd	frr-reload     frrinit.sh    nhrpd   pbrd   ripngd   watchfrr.sh
bgpd	frr-reload.py  isisd	     ospf6d  pimd   staticd  zebra
root@sonic:/home/sonic# docker exec -t -i bgp dpkg --list | grep frr
ii  frr            6.0.2-201901 amd64        FRRouting suite of internet proto
ii  frr-pythontool 6.0.2-201901 all          FRRouting suite - Python tools
root@sonic:/home/sonic# show version
SONiC Software Version: SONiC.add_frr_reload.0-dirty-20190128.181135
Distribution: Debian 9.7
Kernel: 4.9.0-8-amd64
Build commit: 0777bc0
Build date: Tue Jan 29 03:57:43 UTC 2019
```

**Comment by batmancn** [comment]
So, does this PR is for save configure in sonic-frr? #2355 is another way for this.

**Comment by mslocrian** [comment]
@batmancn,

This is for a different use case. @nikos-github has done some great work in splitting out the frr config from something that is managed within sonic, to being able to manage it outside. 

The frr-pythontool package is for the latter case. What you can do is place a new frr config, parse it through the frr-reload.py and it only applies the diff of running config to your desired config. When you want to get a config into place without having to touch the bgpd process, then that can be pretty useful.

**Comment by batmancn** [comment]
> @batmancn,
> 
> This is for a different use case. @nikos-github has done some great work in splitting out the frr config from something that is managed within sonic, to being able to manage it outside.
> 
> The frr-pythontool package is for the latter case. What you can do is place a new frr config, parse it through the frr-reload.py and it only applies the diff of running config to your desired config. When you want to get a config into place without having to touch the bgpd process, then that can be pretty useful.

Great, That's cool.

But will new feature support increment config and reboot?
And if there is some SPEC or PR about these work?

**Comment by mslocrian** [comment]
The PR for the spit config can be found here:
https://github.com/Azure/sonic-buildimage/pull/2286

The file/package that I am adding to the container is already part of the FRR software.
https://github.com/FRRouting/frr/releases

The package is already getting built when sonic goes through its own frr build in the sonic-frr repo. I'm just copying that package over for inclusion to the rest of the build process, and getting it added to the deb package list for installation only in the sonic-frr container.

**Comment by nikos-github** [comment]
I'll take a look. This approach is preferred.

File: dockers/docker-fpm-frr/daemons [hunk 1]
```diff
@@ -36,7 +36,7 @@ fabricd=no
 # Check /etc/pam.d/frr if you intend to use "vtysh"!
 #
 vtysh_enable=yes
-zebra_options="  -A 127.0.0.1 -s 90000000 -M fpm"
```
Comment (nikos-github) [inline dockers/docker-fpm-frr/daemons]: Why are you removing this?

Comment (mslocrian) [reply dockers/docker-fpm-frr/daemons]: That actually should not have happened. I can't recall why/how that file would have changed, or perhaps I merged something in which I shouldn't have. My bad!

Thank you and good catch. It's been fixed.

**Comment by nikos-github** [comment]
retest this please

**Comment by mslocrian** [comment]
@nikos-github , I've rebuilt and reinstalled the image. Things are looking good from that perspective.
```
root@sonic:/usr/lib/frr# grep zeb /etc/frr/daemons
# The watchfrr and zebra daemons are always started.
zebra_options="  -A 127.0.0.1 -s 90000000 -M fpm"
root@sonic:/usr/lib/frr# ps uaxww | grep zeb
root        69  0.0  0.0  80092  2732 ?        Ss   16:16   0:03 /usr/lib/frr/watchfrr -d -r /usr/lib/frr/watchfrr.sh restart %s -s /usr/lib/frr/watchfrr.sh start %s -k /usr/lib/frr/watchfrr.sh stop %s zebra bgpd staticd
frr         84  0.0  0.2 411976 37828 ?        Ssl  16:16   0:02 /usr/lib/frr/zebra -d -A 127.0.0.1 -s 90000000 -M fpm
root       589  0.0  0.0  11128   996 pts/1    S+   22:27   0:00 grep zeb
root@sonic:/usr/lib/frr# pwd 
/usr/lib/frr
root@sonic:/usr/lib/frr# ls
babeld	bfdd  bgpd  eigrpd  frr-reload	frr-reload.py  frrcommon.sh  frrinit.sh  isisd	ldpd  nhrpd  ospf6d  ospfd  pbrd  pimd	ripd  ripngd  staticd  watchfrr  watchfrr.sh  zebra
root@sonic:/usr/lib/frr# ls -l frr-reload
-rwxr-xr-x 1 root root 198 Feb 24 04:16 frr-reload
root@sonic:/usr/lib/frr# ls -l frr-reload.py 
-rwxr-xr-x 1 root root 52120 Feb 26 06:25 frr-reload.py
root@sonic:/usr/lib/frr# 
```
Any other testing that you are looking for?

**Comment by nikos-github** [comment]
@mslocrian The retest this was not for you. It was for jenkins to rerun the tests because they were failing when it first run.

**Comment by nikos-github** [comment]
@lguohan Can we merge this after you review?

**Comment by nikos-github** [comment]
@lguohan This is conflicting with my debian stretch frr changes. Are we getting it merged or should I include in my diffs?

**Comment by mslocrian** [comment]
Would be excited to see this included, @lguohan! Thanks for pushing on it @nikos-github

**Comment by lguohan** [comment]
duplicated as #2819

**Pull request closed without merging**